engines/python/setup/djl_python/huggingface.py

```diff
@@ -199,54 +199,73 @@ def initialize(self, properties: dict):
-                parameters[i]["cached_prompt"] = input_map.pop("cached_prompt")
-
-            seed_key = 'seed' if inputs.is_batch() else f'batch_{i}.seed'
-            if item.contains_key(seed_key):
-                seed = parameters[i].get("seed")
-                if not seed:
-                    # set server provided seed if seed is not part of request
-                    parameters[i]["seed"] = item.get_as_string(key=seed_key)
+            try:
+                content_type = item.get_property("Content-Type")
+                input_map = decode(item, content_type)
+                _inputs = input_map.pop("inputs", input_map)
+                if isinstance(_inputs, list):
+                    input_data.extend(_inputs)
+                    input_size.append(len(_inputs))
+                else:
+                    input_data.append(_inputs)
+                    input_size.append(1)
+                if first or self.rolling_batch_type:
+                    parameters.append(input_map.pop("parameters", {}))
+                    first = False
+                else:
+                    if parameters != input_map.pop("parameters", {}):
+                        raise ValueError(
+                            "In order to enable dynamic batching, all input batches must have the same parameters"
+                        )
+                if "cached_prompt" in input_map:
+                    parameters[i]["cached_prompt"] = input_map.pop(
+                        "cached_prompt")
+
+                seed_key = 'seed' if inputs.is_batch() else f'batch_{i}.seed'
+                if item.contains_key(seed_key):
+                    seed = parameters[i].get("seed")
+                    if not seed:
+                        # set server provided seed if seed is not part of request
+                        parameters[i]["seed"] = item.get_as_string(
+                            key=seed_key)
+            except Exception:  # pylint: disable=broad-except
+                logging.exception("Parse input failed")
+                errors[i] = str(e)
+
+        if len(input_data) == 0:
+            raise ValueError("All batch inputs are invalid")
+
+        return input_data, input_size, parameters, errors, batch
 
+    def inference(self, inputs):
         outputs = Output()
 
+        input_data, input_size, parameters, errors, batch = self.parse_inputs(inputs)
         if self.rolling_batch_type:
             if inputs.get_property("reset_rollingbatch"):
                 self.rolling_batch.reset()
             result = self.rolling_batch.inference(input_data, parameters)
-            for i in range(inputs.get_batch_size()):
-                outputs.add(result[i], key="data", batch_index=i)
+            for i in range(len(input_data)):
+                err = errors.get(i)
+                if err:
+                    outputs.add(err, key="data", batch_index=i)
+                else:
+                    outputs.add(result[i], key="data", batch_index=i)
```

Comment:
i is not the right index, probably we need to +1 everytime we insert a new value. Or use iterator to iter a value.